### DISCUSSION TITLE
Plotting prior vs posterior

### DISCUSSION THREAD

**Discussion by wlandau** [openpharma/brms.mmrm#127] [Q&A] [answered]
I just presented `brms.mmrm` to an internal stats group at my workplace, and I got questions about how to plot the prior vs the posterior, and how to plot the prior on its own. I believe `brms` and `tidybayes` may have solutions for this, but it would be good to check and to see if `brms.mmrm` should jump in and make this easier.

**Comment by yonicd** [comment, marked as answer]
> I just presented `brms.mmrm` to an internal stats group at my workplace, and I got questions about how to plot the prior vs the posterior, and how to plot the prior on its own. I believe `brms` and `tidybayes` may have solutions for this, but it would be good to check and to see if `brms.mmrm` should jump in and make this easier.

{ggdist} should be good for this task.

Eg: https://discourse.mc-stan.org/t/overlaying-prior-curves-to-multiple-posteriors-within-the-same-plot/29548

Noting that the answer is from '22 so the prior mechanism is a bit more polished now in the pkg.

**Reply by wlandau** [reply]
Nice! This is easier than I thought. Added to the usage vignette in 4a1643dd636abc7e9b55159a44e7e890720a83d2.

```r
library(distributional)
library(ggdist)
library(ggplot2)
library(posterior)

prior <- dist_student_t(3, 1.9, 11.8)
posterior <- as_draws_df(model)

ggplot() +
  stat_halfeye(aes(x = b_Intercept), data = posterior) +
  stat_slab(aes(xdist = prior), color = "blue", fill = NA) +
  scale_thickness_shared()
```

![priorvsposterior-1 copy](https://github.com/user-attachments/assets/561681b1-f21d-4aaa-8ef1-b4afeb90c151)
